spec/support/authentication_helpers.rb

```diff
@@ -5,6 +5,38 @@ def authenticate!(user_id='9999', is_panoptes_admin: false)
     allow(controller).to receive(:client).and_return(user_client(user_id, is_panoptes_admin))
   end
 
+  def authenticate_with_membership!(classification_user_group, memberships, is_panoptes_admin: false)
+    allow(controller).to receive(:client).and_return(user_client_with_membership(classification_user_group, memberships, is_panoptes_admin))
+  end
+
+  def panoptes_application_client
+    return @panoptes_application_client if @panoptes_application_client
+
+    @panoptes_application_client = instance_double(Panoptes::Client)
+
+    @panoptes_application_client
+  end
+
+  def user_client_with_membership(classification_user_group, memberships, is_panoptes_admin)
+    return @user_client_with_membership if @user_client_with_membership
+
+    me_hash = {
+      'id' => classification_user_group.user_id,
+      'login' => 'login',
+      'display_name' => 'display_name',
+      'admin' => is_panoptes_admin
+    }
+    memberships_url = "/memberships?user_id=#{classification_user_group.user_id}&user_group_id=#{classification_user_group.user_group_id}"
+
+    @user_client_with_membership = double(Panoptes::Client, me: me_hash).tap do |client|
+      allow(client).to receive(:is_a?).and_return(false)
```

Comment:
What is this protecting against? Does something ask the client this somewhere?

Reply:
Yeah..we don't need this line. We are not asking the the client this anywhere.